ui/store/actions.ts

```diff
@@ -1613,6 +1614,10 @@ export function updateMetamaskState(
           gasEstimateType: newState.gasEstimateType,
         },
       });
+      updateTxValueIfMaxEthSettled(dispatch, state, {
```

Comment:
I appreciate we're trying to piggy back on the gas fee estimate change, but is it best to add it to this low level and already large action?

Would this detection and update logic be better placed in a component where we can map the redux state cleanly and already have access to all the state we need, such as `ConfirmTransactionBase`?

Reply:
Done in https://github.com/MetaMask/metamask-extension/commit/f362be605aa5deda59e74b27bfe810a05b60db8e